modules/unrulyBidAdapter.js

```diff
@@ -1,137 +1,225 @@
-
-      configureUniversalTag(exchangeRenderer);
-      configureRendererQueue();
-
-      const rendererInstance = Renderer.install(Object.assign({}, exchangeRenderer, { callback: () => {} }));
-      return { rendererInstance, serverBid };
-    })
-    .map(
-      ({rendererInstance, serverBid}) => {
-        const prebidBid = serverResponseToBid(serverBid, rendererInstance);
-
-        const rendererConfig = Object.assign(
-          {},
-          prebidBid,
-          {
-            renderer: rendererInstance,
-            adUnitCode: serverBid.ext.adUnitCode
-          }
-        );
-
-        rendererInstance.setRender(() => { notifyRenderer(rendererConfig) });
-
-        return prebidBid;
+const addBidFloorInfo = (validBid) => {
+  Object.keys(validBid.mediaTypes).forEach((key) => {
+    let floor;
+    if (typeof validBid.getFloor === 'function') {
+      floor = validBid.getFloor({
+        currency: 'USD',
+        mediaType: key,
+        size: '*'
+      }).floor || 0;
+    } else {
+      floor = validBid.params.floor || 0;
+    }
+
+    validBid.mediaTypes[key].floor = floor;
+  });
+};
+
+const RemoveDuplicateSizes = (validBid) => {
+  let bannerMediaType = utils.deepAccess(validBid, 'mediaTypes.banner');
+  if (bannerMediaType) {
+    let seenSizes = {};
+    let newSizesArray = [];
+    bannerMediaType.sizes.forEach((size) => {
+      if (!seenSizes[size.toString()]) {
+        seenSizes[size.toString()] = true;
+        newSizesArray.push(size);
       }
-    );
+    });
 
-export const adapter = {
-  code: 'unruly',
-  supportedMediaTypes: [ VIDEO ],
-  isBidRequestValid: function(bid) {
-    if (!bid) return false;
+    bannerMediaType.sizes = newSizesArray;
+  }
+};
 
-    const context = utils.deepAccess(bid, 'mediaTypes.video.context');
+const getRequests = (conf, validBidRequests, bidderRequest) => {
+  const {bids, bidderRequestId, auctionId, bidderCode, ...bidderRequestData} = bidderRequest;
+  const invalidBidsCount = bidderRequest.bids.length - validBidRequests.length;
+  let requestBySiteId = {};
 
-    return bid.mediaType === 'video' || context === 'outstream';
-  },
+  validBidRequests.forEach((validBid) => {
+    const currSiteId = validBid.params.siteId;
+    addBidFloorInfo(validBid);
+    RemoveDuplicateSizes(validBid);
+    requestBySiteId[currSiteId] = requestBySiteId[currSiteId] || [];
+    requestBySiteId[currSiteId].push(validBid);
+  });
 
-  buildRequests: function(validBidRequests, bidderRequest) {
-    const url = 'https://targeting.unrulymedia.com/prebid';
-    const method = 'POST';
-    const data = {
-      bidRequests: validBidRequests,
-      bidderRequest
-    };
-    const options = { contentType: 'text/plain' };
+  let request = [];
 
-    return {
-      url,
-      method,
-      data,
-      options
+  Object.keys(requestBySiteId).forEach((key) => {
+    let data = {
+      bidderRequest: Object.assign({}, {bids: requestBySiteId[key], invalidBidsCount, ...bidderRequestData})
```

Comment:
I am all for using the object spread syntax but just letting you know it won't work in IE11.

Reply:
Looks like it does get transpiled to es5, so I guess, it's fine.